src/concurrency/thread.rs

```diff
@@ -513,6 +498,15 @@ impl<'tcx> ThreadManager<'tcx> {
         }
     }
 
+    pub fn thread_id_try_from(&self, id: impl TryInto<u32>) -> Result<ThreadId, ThreadNotFound> {
+        if let Ok(id) = id.try_into()
+            && id.try_into().unwrap_or(usize::MAX) < self.threads.len()
```

Comment:
The second `try_into` is converting `u32` to `usize`, right?
```suggestion
            && usize::try_from(id).is_some_and(|id| id < self.threads.len())
```